cloud/aws/aws_s3.cc

```diff
@@ -818,14 +819,80 @@ Status S3StorageProvider::CopyCloudObject(const std::string& bucket_name_src,
   return st;
 }
 
+namespace {
+class WritableFileStreamBuf : public std::streambuf {
```

Comment:
@igorcanadi can you have some comments explaining the necessity of this? I understand the overall point of using DIRECT_IO, but not sure what this particular class `WritableFileStreamBuf` is for :D